flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/ResourceRequirementsRetrieverTest.java

```diff
@@ -0,0 +1,207 @@
+		setUp(ResourceSpec.UNKNOWN);
+
+		for (SlotSharingGroup slotSharingGroup : slotSharingGroups) {
+			assertEquals(
+				ResourceProfile.UNKNOWN,
+				resourceRequirementsRetriever.getSlotSharingGroupResourceRequirement(slotSharingGroup.getSlotSharingGroupId()));
+		}
+	}
+
+	private void setUp(final ResourceSpec resources) throws Exception {
+		slotSharingGroups = Arrays.asList(new SlotSharingGroup(), new SlotSharingGroup());
+
+		jobGraph = createJobGraph(slotSharingGroups, resources);
+
+		executionGraph = TestingExecutionGraphBuilder.newBuilder().setJobGraph(jobGraph).build();
+
+		resourceRequirementsRetriever = new ResourceRequirementsRetriever(executionGraph.getAllVertices(), SHUFFLE_MASTER);
+	}
+
+	/**
+	 * Create a job graph for testing.
+	 * The parallelism is 4/5/6 for source/map/sink.
+	 * source and map are in the same slot sharing group. sink is in an individual group.
+	 * map is POINTWISE connected to source. sink is ALL_TO_ALL connected to map.
+	 */
+	private static JobGraph createJobGraph(final List<SlotSharingGroup> slotSharingGroups, final ResourceSpec resources) {
+		final JobVertex source = new JobVertex("source");
+		source.setInvokableClass(NoOpInvokable.class);
+		source.setParallelism(4);
+		source.setSlotSharingGroup(slotSharingGroups.get(0));
+		source.setResources(resources, resources);
+
+		final JobVertex map = new JobVertex("map");
+		map.setInvokableClass(NoOpInvokable.class);
+		map.setParallelism(5);
+		map.setSlotSharingGroup(slotSharingGroups.get(0));
+		map.setResources(resources, resources);
+
+		final JobVertex sink = new JobVertex("sink");
+		sink.setInvokableClass(NoOpInvokable.class);
+		sink.setParallelism(6);
+		sink.setSlotSharingGroup(slotSharingGroups.get(1));
+		sink.setResources(resources, resources);
+
+		map.connectNewDataSetAsInput(source, DistributionPattern.POINTWISE, ResultPartitionType.PIPELINED);
+		sink.connectNewDataSetAsInput(map, DistributionPattern.ALL_TO_ALL, ResultPartitionType.PIPELINED);
+
+		return new JobGraph(source, map, sink);
+	}
+
+	/**
+	 * Test implementation of {@link ShuffleMaster}.
+	 * The shuffle memory size for a task is {@code totalChannels * OFFSET + totalResultSubpartitions}。
+	 */
+	private static class TestShuffleMaster implements ShuffleMaster<ShuffleDescriptor> {
+
+		@Override
+		public CompletableFuture<ShuffleDescriptor> registerPartitionWithProducer(
+				final PartitionDescriptor partitionDescriptor,
+				final ProducerDescriptor producerDescriptor) {
+
+			return null;
+		}
+
+		@Override
+		public void releasePartitionExternally(final ShuffleDescriptor shuffleDescriptor) {
+		}
+
+		@Override
+		public MemorySize getShuffleMemoryForTask(final TaskInputsOutputsDescriptor taskInputsOutputsDescriptor) {
+			final long size = getInputChannelsOfLastTaskInputsOutputsDescriptor(taskInputsOutputsDescriptor) * OFFSET
+				+ getResultSubpartitionsOfLastTaskInputsOutputsDescriptor(taskInputsOutputsDescriptor);
+			return new MemorySize(size);
+		}
```

Comment:
We still need to implement a `ShuffleMaster` to create `ResourceRequirementsRetriever`.
It should be not too bad since there are only 2 other interfaces.